osc/osc.go

```
@@ -926,79 +928,55 @@ func writeBlob(data []byte, buf *bytes.Buffer) (int, error) {
-// bytes are removed from the reader.
-func readPaddedString(reader *bufio.Reader) (string, int, error) {
-	// Read the string from the reader
-	str, err := reader.ReadString(0)
+// readPaddedString reads a padded string from the given buffer. The padding
+// bytes are removed from the buffer.
+func readPaddedString(buffer *bytes.Buffer) (string, int, error) {
+	//Read the string from the buffer
+	str, err := buffer.ReadString(0)
 	if err != nil {
 		return "", 0, err
 	}
-	n := len(str)
 
-	// Remove the string delimiter, in order to calculate the right amount
-	// of padding bytes
-	str = str[:len(str)-1]
+	n := len(str)
 
 	// Remove the padding bytes
-	padLen := padBytesNeeded(len(str)) - 1
-	if padLen > 0 {
-		n += padLen
-		padBytes := make([]byte, padLen)
-		if _, err = reader.Read(padBytes); err != nil {
-			return "", 0, err
-		}
-	}
+	padLen := padBytesNeeded(n)
+	n += padLen
+	buffer.Next(padLen)
 
-	return str, n, nil
+	return str[:len(str)-1], n, nil
 }
 
 // writePaddedString writes a string with padding bytes to the a buffer.
 // Returns, the number of written bytes and an error if any.
-func writePaddedString(str string, buf *bytes.Buffer) (int, error) {
+func writePaddedString(str string, buf *bytes.Buffer) int {
 	// Write the string to the buffer
-	n, err := buf.WriteString(str)
-	if err != nil {
-		return 0, err
-	}
+	n, _ := buf.WriteString(str)
+	// Write the null terminator to the buffer as well
+	buf.WriteByte(0)
+	n++
 
 	// Calculate the padding bytes needed and create a buffer for the padding bytes
-	numPadBytes := padBytesNeeded(len(str))
-	if numPadBytes > 0 {
-		padBytes := make([]byte, numPadBytes)
-		// Add the padding bytes to the buffer
-		n, err := buf.Write(padBytes)
-		if err != nil {
-			return 0, err
-		}
-		numPadBytes = n
-	}
+	numPadBytes := padBytesNeeded(n)
+	// Add the padding bytes to the buffer
+	buf.Write(padBytes[:numPadBytes])
 
-	return n + numPadBytes, nil
```

don't need to return an error if we'll never get one, right?